tests/test_hf_mpt_gen.py

```diff
@@ -72,3 +78,86 @@ def test_init_hfhub_mpt(device: str, attn_impl: str):
 
 def test_init_hfhub_mpt_cpu():
     test_init_hfhub_mpt(device='cpu', attn_impl='torch')
+
+EOS_TOKEN_ID = 0
+
+class MockMPTForCausalLM(MPTForCausalLM):
+    """Class that overrides the forward of MPTForCausalLM.
+    """
+    def forward(
+        self,
+        input_ids: torch.LongTensor,
+        past_key_values: Optional[List[Tuple[torch.FloatTensor]]] = None,
+        attention_mask: Optional[torch.ByteTensor] = None,
+        prefix_mask: Optional[torch.ByteTensor] = None,
+        sequence_id: Optional[torch.LongTensor] = None,
+        labels: Optional[torch.LongTensor] = None,
+        return_dict: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        use_cache: Optional[bool] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+    ):
+        result = super().forward(input_ids, past_key_values, attention_mask, prefix_mask, sequence_id, labels, return_dict, output_attentions, output_hidden_states, use_cache, inputs_embeds)
+        # Modify the logits to select the next token.
+        if dist.get_global_rank() == 0:
+            # Rank 0 hits EOS immediately.
+            result.logits[:, :, EOS_TOKEN_ID] = torch.inf
+        else:
+            # Other ranks do not hit EOS.
+            result.logits[:, :, EOS_TOKEN_ID] = -torch.inf
+        return result
+
+def mock_from_config(config: MPTConfig, **_):
+    config_dict = config.to_dict()
+    config = MPTConfig.from_dict(config_dict)
```

Comment:
this seemed really odd to me that I would need to do this extra step of re-converting the config from a dictionary...Otherwise, it resulted in slightly different model behavior.

Reply:
Actually, the original config here is the MPTConfig class downloaded by transformers...so different from the one in the repo. So maybe there's been some differences between these classes?

Reply:
you likely need to do something like this https://github.com/mosaicml/llm-foundry/blob/aa2ba9f139582348840637ae69945e3140181557/tests/test_onnx.py#L31-L33